queries/http/highlights.scm

```diff
@@ -0,0 +1,33 @@
+; Display errors
+(ERROR) @error
+
+; Comments
+(comment) @comment
+
+(request
+  method: (method) @keyword
+  url: (url) @text.uri)
+
+(header
+  name: (name) @constant
+  value: (value))
+
+; JSON
+(json_pair
+  key: (_) @keyword)
```

Comment:
could you streamline this with the JSON highlighting. They used `@label` for key which may not be ideal either, but it should be consistent between the two.

Reply:
It looks like this parser is directly embedding rules from the JSON parser. It could also just have extended this grammar. If the JSON nodes have identical names as in real JSON you could also just import the JSON highlights (see how `cpp/highlights.scm` does this with C highlights.

Reply:
and maybe this could use an injection instead too. I guess most of  the http bodies are json, but not always. The injection could make use of the Content-Type header.

Reply:
I am not quite sure how to proceed here, my knowledge of TS is pretty much non existent. @NTBBloodbath, is it something you understand?

Reply:
I also thought about injection but I can understand that with just more 10 lines with the JSON rules, you have a more powerful parser for stand-alone usage. But yes, if it's just for nvim you could also leave JSON for an injected JSON parser.

Reply:
Hi, sorry for the late response!

I think injections are fine, I made this parser for nvim so doing the injections could be a better way. I saw some examples (also thanks for referencing the cpp highlights file) and I'll be doing some tests today to completely inherit and inject the json grammar and highlights.

I'll let you know once it works and is pushed to upstream :)

Reply:
@NTBBloodbath there's actually two possibilities. Either you could inherit JSON, then it will be embedded in your parser or you just parse a placeholder node. Then, nvim can take care of using another parser for that range. But anyways: JSON is quite small and will probably never change. So it could just be over-engineering and any solution you take will be fine.

Reply:
Hey, got some good news. Already did the JSON parser inheritance in both grammar and highlights and is working like a charm :)

Pushed the changes to upstream so the only thing left would be updating the highlights query here, am I right? Also thanks again for the help and feedback about the parser :heart:

Regards

Reply:
I just pushed that, hope this is the right way.

You guys are amazing!